agent/agent.go

```diff
@@ -174,6 +178,9 @@ func _main() int {
 	// Start sending events to the backend
 	go eventhandler.HandleEngineEvents(taskEngine, client, stateManager)
 
+	// Start metrics session in a go routine
+	go startMetricsSession(containerInstanceArn, credentialProvider, cfg, true, client, taskEngine)
```

Comment:
The 'acceptInvalidCert' parameter should be taken from flags, not hardcoded true. Also, this is enough arguments it could make sense to break out as an input struct for readability (yes, I know the rest of main doesn't).